dlt/common/configuration/inject.py

```diff
@@ -167,12 +172,48 @@ def _wrap(*args: Any, **kwargs: Any) -> Any:
                 bound_args.arguments[kwargs_arg.name].update(resolved_params)
                 bound_args.arguments[kwargs_arg.name][_LAST_DLT_CONFIG] = config
                 bound_args.arguments[kwargs_arg.name][_ORIGINAL_ARGS] = (args, kwargs)
+
+        def create_resolved_partial() -> Any:
+            # creates a pre-resolved partial of the decorated function
+            empty_bound_args = sig.bind_partial()
+            config = resolve_config(empty_bound_args)
+
+            # TODO: do some checks, for example fail if there is a spec arg
+
+            def creator(*args: Any, **kwargs: Any) -> Any:
+                nonlocal config
+
+                # we can still overwrite the config
+                if _LAST_DLT_CONFIG in kwargs:
+                    config = last_config(**kwargs)
+
+                # call the function with the pre-resolved config
+                bound_args = sig.bind(*args, **kwargs)
+                update_bound_args(bound_args, config, args, kwargs)
+                return f(*bound_args.args, **bound_args.kwargs)
+
+            return creator
+
+        @wraps(f)
+        def _wrap(*args: Any, **kwargs: Any) -> Any:
```

Comment:
1. we must sure that config is resolved at run time (when the inner function is actually called)
2. we do not want to cache here in interactive mode

Reply:
1. This is the case
2. I'm not sure what interactive mode is exactly?

Reply:
In any case, in the tests you can see that you can create a pipeline, then define env vars and run the pipeline without failure.